arch/xtensa/src/common/xtensa_lwl_console.c

```diff
@@ -0,0 +1,314 @@
+/****************************************************************************
+ * arch/xtensa/src/common/xtensa_lwl_console.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <sys/types.h>
+#include <errno.h>
+#include <debug.h>
+
+#include <nuttx/arch.h>
+#include <nuttx/fs/fs.h>
+#include <syscall.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* Lightweight Link (lwl)
```

Comment:
@acassis the protocol and implementation don't depend on arch or chip, why not move lwl_console.c to nuttxs/drivers folder?

Reply:
Ok, I agree! Do think it should be at drivers/lwl_console.c or should we create a directory to put it? If so, which name should we use? drivers/debug/ drivers/trace/ drivers/link/ etc ?

Reply:
the most console like drivers put into drivers/serial or drivers/syslog, but I am fine with other locations. or we create drivers/misc and then move lwl_console.c, zero.c, null.c and ramdisk.c to there.